jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverter.java

```diff
@@ -38,1082 +38,1265 @@
-            replaceComputedLane(computedLane);
-         }
+    }
+  }
+
+  /**
+   * Replaces shape point set.
+   *
+   * @param shapePointSet JsonNode representing the shape point set
+   */
+  public static void replaceShapePointSet(JsonNode shapePointSet) {
+    // shape point set contains:
+    // anchor
+    // lane width
+    // directionality
+    // node list
+
+    ObjectNode updatedNode = (ObjectNode) shapePointSet;
+
+    // replace anchor
+    if (updatedNode.has(ANCHOR)) {
+      JsonUtils.addNode(updatedNode, ANCHOR,
+          Position3DBuilder.dsrcPosition3D(
+              Position3DBuilder.odePosition3D(updatedNode.get(ANCHOR))));
+    }
+
+    // replace lane width
+    if (updatedNode.has(LANE_WIDTH)) {
+      updatedNode.put(LANE_WIDTH,
+          LaneWidthBuilder.laneWidth(JsonUtils.decimalValue(updatedNode.get(LANE_WIDTH))));
+    }
+
+    // replace directionality
+    if (updatedNode.has(DIRECTIONALITY)) {
+      JsonNode directionality = updatedNode.get(DIRECTIONALITY);
+      String enumString =
+          CommonUtils.enumToString(DirectionOfUseEnum.class, directionality.asText());
+      if (enumString != null) {
+        updatedNode.set(DIRECTIONALITY, JsonUtils.newNode().put(enumString, EMPTY_FIELD_FLAG));
       }
-   }
-
-   public static void replaceComputedLane(JsonNode jsonNode) {
-      ObjectNode updatedNode = (ObjectNode) jsonNode;
-
-      // Nothing to do for referenceLaneId LaneID
-
-      // offsetXaxis CHOICE
-      // small DrivenLineOffsetSm,
-      // large DrivenLineOffsetLg
-      //
-      replaceScale(updatedNode, OFFSET_X_AXIS);
-
-      // offsetYaxis CHOICE
-      // small DrivenLineOffsetSm,
-      // large DrivenLineOffsetLg
-      //
-      replaceScale(updatedNode, OFFSET_Y_AXIS);
-
-      // rotateXY Angle OPTIONAL
-      if (updatedNode.has(ROTATE_XY)) {
-         updatedNode.put(ROTATE_XY, AngleBuilder.angle(JsonUtils.decimalValue(updatedNode.get(ROTATE_XY))));
+    }
+
+    // replace node list
+    if (updatedNode.has(NODE_LIST)) {
+      ObjectNode nodeList = (ObjectNode) updatedNode.get(NODE_LIST);
+      if (nodeList.has(NODES)) {
+        ArrayNode nodes = transformNodeSetXY(nodeList.get(NODES));
+        nodeList.set(NODES, nodes);
+      } else if (nodeList.has(COMPUTED)) {
+        JsonNode computedLane = nodeList.get(COMPUTED);
+        replaceComputedLane(computedLane);
       }
-
-      // scaleXaxis Scale-B12 OPTIONAL
-      if (updatedNode.has(SCALE_X_AXIS)) {
-         updatedNode.put(SCALE_X_AXIS, ScaleB12Builder.scaleB12(JsonUtils.decimalValue(updatedNode.get(SCALE_X_AXIS))));
+    }
+  }
+
+  /**
+   * Replaces computed lane.
+   *
+   * @param jsonNode JsonNode representing the computed lane
+   */
+  public static void replaceComputedLane(JsonNode jsonNode) {
+    ObjectNode updatedNode = (ObjectNode) jsonNode;
+
+    // Nothing to do for referenceLaneId LaneID
+
+    // offsetXaxis CHOICE
+    // small DrivenLineOffsetSm,
+    // large DrivenLineOffsetLg
+    //

```

Comment:
(non-blocking) Can we remove these empty lines of comments?

Reply:
Yup! I've removed all the unnecessary empty comments from this class.